examples/arduino-uno/src/bin/uno-timer.rs

```diff
@@ -0,0 +1,136 @@
+#![no_std]
+#![no_main]
+#![feature(abi_avr_interrupt)]
+
+/*
+ This is derived from Rahix' comment to
+https://github.com/Rahix/avr-hal/issues/75
+and then modernized to account for API drift since 2020
+
+*/
+
+use arduino_hal::hal::port::PB5;
+use arduino_hal::port::mode::Output;
+use arduino_hal::port::Pin;
+use arduino_hal::prelude::*;
+use avr_device::atmega328p::tc1::tccr1b::CS1_A;
+use avr_device::atmega328p::TC1;
+use core::mem;
+use panic_halt as _;
+use ufmt::{uWrite, uwriteln};
+
+struct InterruptState<LED> {
+    blinker: Pin<Output, LED>,
+}
```

Comment:
I think it would be nicer to get rid of the `<LED>` type parameter and instead use a "dynamic Pin" as described in ["Downgrading"](https://rahix.github.io/avr-hal/avr_hal_generic/port/struct.Pin.html#downgrading).  The type would then be

```rust
struct InterruptState {
    blinker: Pin<Output>,
}
```